### PR TITLE
INTEL MKL: Fixing a Bug in Quantized Convolution

### PR DESCRIPTION
**Pull request by mahmoud-abuzaina**
This PR fixes a bug in MKL quantized convolution. It also includes clang-format changes.

### CODE REVIEW

File: tensorflow/core/kernels/mkl_conv_ops.cc [hunk 1]
```diff
@@ -1044,8 +1042,8 @@ class MklConvOp : public OpKernel {
       AllocateOutputTensor(context, *conv_fwd_pd, dst_dims_mkl_order, tf_fmt,
                            &dst_tensor);
       Tensor* filter_out_tensor = nullptr;
-      if (typeid(Tinput) == typeid(float) && typeid(Tfilter) == typeid(float) &&
-          typeid(Toutput) == typeid(float)) {
+      if (typeid(Tinput) == typeid(float)&&typeid(Tfilter) ==
+          typeid(float)&&typeid(Toutput) == typeid(float)) {
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_conv_ops.cc]: This formatting looks weird. Did you use a different formatter than `clang-format`? Could you please try applying `clang-format`?

File: tensorflow/core/kernels/mkl_conv_ops.cc [hunk 2]
```diff
@@ -323,7 +323,7 @@ class MklConvFwdPrimitiveFactory : public MklPrimitiveFactory<T> {
       const MklConvFwdParams& convFwdDims, bool do_not_cache) {
     MklConvFwdPrimitive<T, Tinput, Tfilter, Tbias, Toutput>* conv_fwd = nullptr;
 
-    if (do_not_cache) { /* Always create new primitive */
+    if (do_not_cache) {/* Always create new primitive */
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_conv_ops.cc]: I think the space between `{` and `/*` should stay. Could you please try applying `clang-format`?

File: tensorflow/core/kernels/mkl_conv_ops.cc [hunk 3]
```diff
@@ -1067,8 +1065,14 @@ class MklConvOp : public OpKernel {
...
-        filter.CheckReorderToOpMem(conv_fwd_pd.get()->weights_primitive_desc(),
-                                   filter.GetTensorBuffer(filter_out_tensor));
+        if (filter_out_tensor == nullptr) {
+          filter.CheckReorderToOpMem(
+              conv_fwd_pd.get()->weights_primitive_desc());
+        } else {
+          filter.CheckReorderToOpMem(
+              conv_fwd_pd.get()->weights_primitive_desc(),
+              filter.GetTensorBuffer(filter_out_tensor));
+        }
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_conv_ops.cc]: Is this the only place for the bug fix? Just wanted to make sure.

File: tensorflow/core/kernels/mkl_conv_ops.cc [hunk 4]
```diff
@@ -1784,34 +1789,31 @@ REGISTER_KERNEL_BUILDER(
...
-          .TypeConstraint<float>("T")                                  \
-          .Label(mkl_op_registry::kMklOpLabel),                        \
-      MklConvOp<CPUDevice, float, float, float, float, float, true>);  \
-  REGISTER_KERNEL_BUILDER(Name("__MklDummyConv2DWithBias")             \
-                              .Device(DEVICE_CPU)                      \
-                              .TypeConstraint<T>("T")                  \
-                              .Label(mkl_op_registry::kMklOpLabel),    \
+#define REGISTER_MKL_CPU_2D(T)                                                \
+  REGISTER_KERNEL_BUILDER(                                                    \
+      Name("_MklConv2D").Device(DEVICE_CPU).TypeConstraint<float>("T").Label( \
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_conv_ops.cc]: I think the original formatting looks better. `clang-format` will fix this.

File: tensorflow/core/kernels/mkl_conv_ops.cc [hunk 5]
```diff
@@ -1784,34 +1789,31 @@ REGISTER_KERNEL_BUILDER(
...
-#define REGISTER_MKL_CPU_3D(T)                                      \
-  REGISTER_KERNEL_BUILDER(Name("_MklConv3D")                        \
-                              .Device(DEVICE_CPU)                   \
-                              .TypeConstraint<T>("T")               \
-                              .Label(mkl_op_registry::kMklOpLabel), \
-                          MklConvOp<CPUDevice, T, T, T, T, T, false>);
+#define REGISTER_MKL_CPU_3D(T)                                            \
+  REGISTER_KERNEL_BUILDER(                                                \
+      Name("_MklConv3D").Device(DEVICE_CPU).TypeConstraint<T>("T").Label( \
+          mkl_op_registry::kMklOpLabel),                                  \
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_conv_ops.cc]: Please fix the alignment with `clang-format`.

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR! The bug fix is hard to see among all the formatting changes. Are the changes in lines 1068-1075 the only changes for the fix? The formatting also looks weird. Please apply `clang-format`.

File: tensorflow/core/kernels/mkl_conv_ops.cc (hunk 1 above)
Comment (mahmoud-abuzaina) [reply tensorflow/core/kernels/mkl_conv_ops.cc]: @penpornk Do you know if there is a specific version of clang-format that we need to use? I use 3.4.2 version with the command: 
clang-format <file_name> --style=google

Comment (penpornk) [reply tensorflow/core/kernels/mkl_conv_ops.cc]: What about without `--style=google`?

Comment (mahmoud-abuzaina) [reply tensorflow/core/kernels/mkl_conv_ops.cc]: @penpornk Thanks! It looks like I was running an old version of clang-format. I re-ran a newer version. Please have a look. The actual bug fix is at line: https://github.com/tensorflow/tensorflow/pull/24021/files#diff-192dfeafbdd684934bdb0dfa8983674aR1064
I will run the newer version for #23724 as well.

**Comment by penpornk** [comment]
@mahmoud-abuzaina The new `clang-format` looks great! Thank you! (Sorry I couldn't tell you the version earlier. I don't know what version we are base on.)

File: tensorflow/core/kernels/mkl_conv_ops.cc (hunk 1 above)
Comment (penpornk) [reply tensorflow/core/kernels/mkl_conv_ops.cc]: Thank you very much! This looks great! (Sorry I couldn't tell you the version earlier. I don't know what version ours is based on.)

**Review by penpornk — APPROVED** [review]
Thank you for the changes!

**Pull request merged**